torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,326 @@
+import collections
+import dataclasses
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+from typing import List, Tuple
+
+import torch
+
+
+@dataclasses.dataclass(repr=False, eq=False, frozen=True)
+class CallgrindStats(object):
+    stmt: str
+    setup: str
+    number: int
+    num_threads: int
+    built_with_debug_symbols: bool
+    baseline_inclusive_stats: List[Tuple[int, str]]
```

Comment:
frozen isn't deep immutability; if you want deep immutability declare these as `Sequence` instead of `List` so that the mutators don't typecheck

Reply:
Switched everything to tuples.